packages/datadog-plugin-kafkajs/src/producer.js

```diff
@@ -1,12 +1,51 @@
 'use strict'
 
 const ProducerPlugin = require('../../dd-trace/src/plugins/producer')
+const { getPathwayHash, encodePathwayContext, decodePathwayContext } = require('./hash')
 
 class KafkajsProducerPlugin extends ProducerPlugin {
   static get id () { return 'kafkajs' }
   static get operation () { return 'produce' }
 
   start ({ topic, messages }) {
+    const env = this.tracer._env
+    const service = this.tracer._service
+
+    let pathwayCtx
+    if (this.config.dsmEnabled !== 'disabled') {
+      const active = this.activeSpan
+      let parentHash
+      let originTimestamp
+      let prevTimestamp
+      const currentTimestamp = new Date().now() * 1000 // nanoseconds
```

Comment:
On second look, I believe this is a microsecond instead of a nanosecond -- `Date.now()` is in milliseconds and microsecond is 1/1000 of a ms.

Reply:
Specifically, the `1000` is correct, just that the comment should be changed to say "// microseconds"